internal/testrunner/runners/system/tester.go

```diff
@@ -1592,17 +1620,25 @@ func (r *tester) validateTestScenario(ctx context.Context, result *testrunner.Re
 	}
 
 	if expectedDatasets == nil {
-		var expectedDataset string
-		if ds := r.testFolder.DataStream; ds != "" {
-			expectedDataset = getDataStreamDataset(*r.pkgManifest, *r.dataStreamManifest)
-		} else {
-			expectedDataset = r.pkgManifest.Name + "." + scenario.policyTemplateName
```

Comment:
These are the same datasets set in `createPackageDatastream` function, so it can be retrieved directly from the kibanaDatastream field set during the preparation of the scenario.

Reply:
This can be changed as:
```golang
		// get dataset directly from package policy added when preparing the scenario
		expectedDataset := scenario.kibanaDataStream.Inputs[0].Streams[0].DataStream.Dataset
		if r.pkgManifest.Type == "input" {
			if scenario.policyTemplateInput == otelCollectorInputName {
				// Input packages whose input is `otelcol` must add the `.otel` suffix
				// Example: httpcheck.metrics.otel
				expectedDataset += "." + otelSuffixDataset
			}
		}
		expectedDatasets = []string{expectedDataset}
```